cni/pkg/nodeagent/informers.go

```diff
@@ -82,12 +82,19 @@ func setupHandlers(ctx context.Context, kubeClient kube.Client, dataplane MeshDa
 	return s
 }
 
+// GetPodIfAmbient looks up a pod. It returns:
+// * An error if the pod cannot be found
+// * nil if the pod is found, but does not have ambient enabled
```

Comment:
Is this just using `nil, nil` as a sentinel value? Should we just return a different error value for when the pod is found but isn't configured as we expect?

Reply:
I really don't like checking error values personally, its super easy to break accidentally by helpfully wrapping it with more context, etc